hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoDeleteHandler.java

```diff
@@ -100,6 +101,12 @@ public void handle(@NonNull final HandleContext context) {
         final var updatedDeleteAccount = accountStore.get(op.deleteAccountID());
         validateTrue(updatedDeleteAccount != null, ACCOUNT_ID_DOES_NOT_EXIST);
         accountStore.put(updatedDeleteAccount.copyBuilder().deleted(true).build());
+
+        // add the transfer account for this deleted account to record builder.
+        // This is needed while computing staking rewards. In the future it will also be added
+        // to the transaction record exported to mirror node.
+        final var cryptoDeleteBuilder = context.recordBuilder(CryptoDeleteRecordBuilder.class);
+        cryptoDeleteBuilder.addBeneficiaryForDeletedAccount(op.deleteAccountID(), op.transferAccountID());
```

Comment:
The protobuf for crypto delete says that `deleteAccountID` and `transferAccountID` are `@Nullable`, but the `addBeneficiaryForDeletedAccount(...)` method lists these inputs as `@NonNull`

Reply:
yeah because in `validateSemantics` we fail if `deleteAccount` is null or `transferAccount` is null.